src/number.js

```diff
@@ -79,16 +82,21 @@ inherits(NumberSchema, MixedSchema, {
     });
   },
 
-  positive(msg = locale.positive) {
-    return this.min(0, msg);
+  positive(msg = null) {
+    return this.min(0, msg, 'number.positive');
```

Comment:
this is a concern of mine, this api does not scale or compose well. Ideally we only want one message argument that can somehow be evaluated lazily. Perhaps instead of `localePath` pass an object with a custom `toString` that resolved to the string value you want?